### ISSUE TITLE
[eslint-config-react-native-community] Upgrade @typescript-eslint/eslint-plugin

### ISSUE THREAD

**Issue by Barbiero** [facebook/react-native#28293]
Description:

 Same as #28214 but with environment info to satisfy bots. I'm not sure why the original poster closed his issue, but upgrading eslint-config to latest plugins would be nice, if only for the parity of packages and to remove install warnings.

React Native version:

```
System:
    OS: Linux 5.3 Ubuntu 19.10 (Eoan Ermine)
    CPU: (4) x64 Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
    Memory: 7.52 GB / 19.43 GB
    Shell: 5.7.1 - /usr/bin/zsh
  Binaries:
    Node: 12.16.1 - ~/.nvm/versions/node/v12.16.1/bin/node
    Yarn: Not Found
    npm: 6.13.4 - ~/.nvm/versions/node/v12.16.1/bin/npm
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce

1. Install `react-native-community/eslint-config` on a fresh RN project
2. Result:

```
npm WARN @typescript-eslint/eslint-plugin@1.13.0 requires a peer of eslint@^5.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @typescript-eslint/parser@1.13.0 requires a peer of eslint@^5.0.0 but none is installed. You must install peer dependencies yourself.
```

## Expected Results

A fresh RN project uses ESLint 6, so it stands to reason that the eslint-config package should be compatible.

**Comment by matt-oakes** [comment]
Thanks for raising this @Barbiero. If you would be able to send over a pull request for it, then I would be happy to review and release a new version.

This can be done independently of the React Native release cycle.

**Comment by matt-oakes** [comment]
I've just published the updated package to NPM as v1.0.0. You should be able to use it now 👍
